src/containers/SignUpForm/SignUpForm.js

```diff
@@ -1,17 +1,105 @@
 import React, { Component } from 'react';
 import { createUser } from '../../utils/queries';
+import { Redirect } from 'react-router-dom';
 
 class SignUpForm extends Component {
   constructor() {
     super();
     this.state = {
-
+      name: '',
+      email: '',
+      passwordOriginal: '',
+      passwordConfirmed: '',
+      status: ''
     };
   }
 
+  handleChange = (event) => {
+    const { id, value } = event.target;
+    this.setState({ [id]: value });
+  }
+
+  handleSubmit = async (event) => {
+    event.preventDefault();
+    const { name, email, passwordOriginal } = this.state;
+    if (this.checkMatchingPassword() && this.checkEmailRegex()) {
+      try {
+        const response = await createUser(name, email, passwordOriginal);
+        this.setState({ status: response.status });
+      } catch (error) {
+        this.setState({ status: 'error' });
+      }
+    }
+  }
+
+  checkEmailRegex = () => {
+    const emailRegex = /[a-z]\w+@[a-z]\w+\.[a-z]\w+/;
+    return emailRegex.test(this.state.email);
+  }
```

Comment:
I really like how this regex functionality allows a more comprehensive email validation than the built-in form validation; I'll use this in my login component as well.